src/env.h

```diff
@@ -340,11 +340,10 @@ constexpr size_t kFsStatsBufferLength = kFsStatsFieldsNumber * 2;
   V(x_forwarded_string, "x-forwarded-for")                                     \
   V(zero_return_string, "ZERO_RETURN")
 
-#define ENVIRONMENT_STRONG_PERSISTENT_PROPERTIES(V)                            \
+#define ENVIRONMENT_STRONG_PERSISTENT_DATA(V)                                  \
```

Comment:
You could merge this with `ENVIRONMENT_STRONG_PERSISTENT_VALUES` - `v8::Value` is a subclass of `v8::Data` too and everything in `ENVIRONMENT_STRONG_PERSISTENT_VALUES` is a subclass of `v8::Value`.

That said, since everything in `ENVIRONMENT_STRONG_PERSISTENT_DATA` is a subclass of `v8::Template` now, you should name it something like `PER_ISOLATE_TEMPLATE_PROPERTIES` and make them per-isolate.

Reply:
Good point about making these `v8::Template`, though conceptually I think these are per-Environment, to add these into the snapshot I'll also need to register the external references for them, which is WIP, and the reason that these can be excluded from the minimal snapshot that we have (which is captured before Environment creation) is that none of these contains external references prior to Environment creation.